sdk/go/common/resource/plugin/host.go

```diff
@@ -343,7 +345,22 @@ func (host *defaultHost) ListAnalyzers() []Analyzer {
 func (host *defaultHost) Provider(pkg tokens.Package, version *semver.Version) (Provider, error) {
 	plugin, err := loadPlugin(host.loadRequests, func() (interface{}, error) {
 		// Try to load and bind to a plugin.
-		plug, err := NewProvider(host, host.ctx, pkg, version, host.runtimeOptions, host.disableProviderPreview)
+
+		result := make(map[string]string)
+		for k, v := range host.config {
```

Comment:
I don't think that's necessary. This will have no impact on any of our providers and it will only send provider key'd config values that would be sent to the default provider process anyway so there's no info leak.

Reply:
Yeah, I agree with Fraser.